### PR TITLE
Text preparation part 1: split 2d rasterization and positioning from 3d mapping

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

File: libnopegl/src/node_text.c [hunk 1]
```diff
@@ -81,12 +81,14 @@ struct text_opts {
 };
 
 struct text_priv {
+    /* characters (fg) */
```
Comment (mbouron) [inline libnopegl/src/node_text.c:84]: nit: no need to specify (fg), imo.

**Review by mbouron — APPROVED** [review]
Other than my remarks, LGTM.

File: libnopegl/src/node_text.c (hunk 1 above)
Comment (ubitux) [reply libnopegl/src/node_text.c]: Okay, dropped (I wanted to make a distinction between the fg and bg pipelines)

File: libnopegl/src/node_text.c [hunk 2]
```diff
@@ -267,12 +267,14 @@ static int update_character_geometries(struct ngl_node *node)
...
 
-        /* quad vertices */
+        /* character position */
         const float chr_corner[3] = {
             corner[0] + width[0] * chr->x + height[0] * chr->y,
             corner[1] + width[1] * chr->x + height[1] * chr->y,
             corner[2] + width[2] * chr->x + height[2] * chr->y,
         };
+
+        /* quad vertices */
```
[the remark this answers is not in the document]
Comment (ubitux) [reply libnopegl/src/node_text.c]: Mmh, these comments aren't actually valuable, I adjusted the commit like this:

```diff
diff --git a/libnopegl/src/node_text.c b/libnopegl/src/node_text.c
index ee389fddf..aa7f12768 100644
--- a/libnopegl/src/node_text.c
+++ b/libnopegl/src/node_text.c
@@ -263,16 +263,16 @@ static int update_character_geometries(struct ngl_node *node)
         const struct char_info *chr = &chars[n];
         float chr_width[3], chr_height[3];

-        /* character dimension */
+        /* character dimension and position */
         ngli_vec3_scale(chr_width, width, chr->w);
         ngli_vec3_scale(chr_height, height, chr->h);
-
-        /* quad vertices */
         const float chr_corner[3] = {
             corner[0] + width[0] * chr->x + height[0] * chr->y,
             corner[1] + width[1] * chr->x + height[1] * chr->y,
             corner[2] + width[2] * chr->x + height[2] * chr->y,
         };
+
+        /* deduce quad vertices from chr_{widht,height,corner} */
         const float chr_vertices[] = {
             C(0),               C(1),               C(2),
             C(0) + W(0),        C(1) + W(1),        C(2) + W(2),
```

File: libnopegl/src/text.c [hunk 3]
```diff
@@ -89,6 +88,8 @@ int ngli_text_init(struct text *s, const struct text_config *cfg)
     if (ret < 0)
         return ret;
 
+    s->texture = s->ctx->font_atlas;
```
[the remark this answers is not in the document]
Comment (ubitux) [reply libnopegl/src/text.c:91]: I can do that in a separate commit if you want but:
1. `atlas_create()` doesn't need the `struct text` context currently so I'll need to pass it down
2. while `s->texture = <something>` is going to stay in the init, `atlas_create()` will likely go away soon™

**Pull request merged**